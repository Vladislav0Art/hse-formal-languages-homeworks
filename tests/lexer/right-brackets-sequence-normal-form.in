start=$A$
$S$ -> $lb$ $B$;
$A$ -> $lb$ $B$ | NULL;
$B$ -> $S$  $C$ | $rb$ $S$ | ')';
$C$ -> $rb$ $S$ | ')';
$lb$ -> '(';
$rb$ -> ')';