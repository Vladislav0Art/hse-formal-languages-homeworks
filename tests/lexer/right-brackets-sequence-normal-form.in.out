LexToken(START,'A',1,0)
LexToken(NON_TERMINAL,'S',2,9)
LexToken(RULE,'->',2,12)
LexToken(NON_TERMINAL,'lb',2,15)
LexToken(NON_TERMINAL,'B',2,19)
LexToken(SEPARATOR,';',2,21)
LexToken(NON_TERMINAL,'A',3,23)
LexToken(RULE,'->',3,26)
LexToken(NON_TERMINAL,'lb',3,29)
LexToken(NON_TERMINAL,'B',3,33)
LexToken(OR,'|',3,36)
LexToken(NULL,'NULL',3,38)
LexToken(SEPARATOR,';',3,42)
LexToken(NON_TERMINAL,'B',4,44)
LexToken(RULE,'->',4,47)
LexToken(NON_TERMINAL,'S',4,50)
LexToken(NON_TERMINAL,'C',4,54)
LexToken(OR,'|',4,57)
LexToken(NON_TERMINAL,'rb',4,59)
LexToken(NON_TERMINAL,'S',4,63)
LexToken(OR,'|',4,66)
LexToken(TERMINAL,')',4,68)
LexToken(SEPARATOR,';',4,71)
LexToken(NON_TERMINAL,'C',5,73)
LexToken(RULE,'->',5,76)
LexToken(NON_TERMINAL,'rb',5,79)
LexToken(NON_TERMINAL,'S',5,83)
LexToken(OR,'|',5,86)
LexToken(TERMINAL,')',5,88)
LexToken(SEPARATOR,';',5,91)
LexToken(NON_TERMINAL,'lb',6,93)
LexToken(RULE,'->',6,97)
LexToken(TERMINAL,'(',6,100)
LexToken(SEPARATOR,';',6,103)
LexToken(NON_TERMINAL,'rb',7,105)
LexToken(RULE,'->',7,109)
LexToken(TERMINAL,')',7,112)
LexToken(SEPARATOR,';',7,115)
